js/event.js

```diff
@@ -1,29 +1,119 @@
-"use strict"
+import { FormData } from "./database.js"
+import { MILISECOND_HOUR, getToday } from "./dateManipulation.js"
+import { selectedDate, storageState } from "./state.js"
+
+const TIMESLOT_LENGTH = 15
 
-const eventButtons_create = document.querySelectorAll('.create');
 const eventModal = document.querySelector("#eventModal")
+const eventForm = document.querySelector("#eventForm")
+const eventButtons_create = document.querySelectorAll('.create');
 const eventButton_cancel = document.querySelector("#event-cancel")
 const eventButton_save = document.querySelector("#event-save")
+const eventDate = document.querySelector('#event-date')
+const startTime = document.querySelector('#event-startTime')
+const endTime = document.querySelector('#event-endTime')
+
+//takes in string in form of 'HH:MM'
+const parseTimeString = (string) => {
+    const result = {}
+    result.hour = parseInt(string.slice(0, 2))
+    result.minutes = parseInt(string.slice(3, 5))
+    return result
+}
+
+const collectFormData = () => {
+
+    const formData = new FormData()
+
+    const inputData = {}
+    const inputs = document.querySelectorAll('[data-key]')
 
-const eventButtons = document.querySelectorAll('#eventModal button')
+    inputs.forEach(input => {
+        if(input.required && !input.value){
+            throw(`Required field (${input.dataset.key}) skipped`)
+        }
+        const key = input.dataset.key
+        const value = input.value
+        inputData[key] = value ? value : null
+    })
+
+    formData.title = inputData.title
+
+    const { startTime : startTimeString, endTime : endTimeString} = inputData
+
+    if(inputData.date && startTimeString && endTimeString){
+
+        const startTime = parseTimeString(startTimeString)
+        const startDate = new Date(inputData.date)
+        startDate.setHours(startTime.hour, startTime.minutes)
+        formData.startDate = Date.parse(startDate)
+    
+        const endTime = parseTimeString(endTimeString)
+        const endDate = new Date(inputData.date)
+        endDate.setHours(endTime.hour, endTime.minutes)
+        formData.endDate = Date.parse(endDate)
+
+        formData.duration = Math.ceil( (formData.endDate - formData.startDate) / (MILISECOND_HOUR/4) )
+        formData.timeSlot = Math.floor( startTime.minutes /  TIMESLOT_LENGTH)
+        formData.cellTimestamp = startDate.setMinutes(0).valueOf()
+    }
 
-eventButtons.forEach(btn => {
-    btn.addEventListener('click', (e) => e.preventDefault())
+    formData.setId()
+    formData.createdAt = getToday().valueOf()
```

Comment:
I can see that formData is implemented like a class. You do "new FormData()". Why not make FormData accept arguments as a class constructor and assign everything at once so object would be valid at once? If you need to set id, it can be gathered using next event id selector - I commented in some other place that maybe it would be more convenient to have a selector function which would calculate next id instead of storing it.